src/backend/access/transam/parallel.c

```diff
@@ -1486,6 +1501,12 @@ ParallelWorkerMain(Datum main_arg)
 	/* Attach to the leader's serializable transaction, if SERIALIZABLE. */
 	AttachSerializableXact(fps->serializable_xact_handle);
 
+
+	/* Hook for babelfish to restore babelfish fixed parallel state */
+	if (MyFixedParallelState->babelfish_context && babelfixedparallelstate_restore_hook)
```

Comment:
Move this above `RestoreUncommittedEnums(uncommittedenumsspace);`

Reply:
Updated in latest rev.